app/src/main/java/com/example/abhishekupadhyay/maphiking/MapsActivity.java

```diff
@@ -0,0 +1,235 @@
+package com.example.abhishekupadhyay.maphiking;
+
+import android.*;
+import android.Manifest;
+import android.app.AlertDialog;
+import android.app.Dialog;
+import android.content.Context;
+import android.content.DialogInterface;
+import android.content.Intent;
+import android.content.pm.PackageManager;
+import android.graphics.Color;
+import android.location.Criteria;
+import android.location.Location;
+import android.location.LocationListener;
+import android.location.LocationManager;
+import android.os.Build;
+import android.provider.Settings;
+import android.support.annotation.NonNull;
+import android.support.v4.app.ActivityCompat;
+import android.support.v4.app.FragmentActivity;
+import android.os.Bundle;
+import android.provider.Settings.Secure;
+import android.support.v4.content.ContextCompat;
+import android.support.v4.content.PermissionChecker;
+import android.util.Log;
+import android.webkit.PermissionRequest;
+import android.widget.Button;
+import android.widget.Toast;
+
+import com.google.android.gms.common.ConnectionResult;
+import com.google.android.gms.common.GooglePlayServicesUtil;
+import com.google.android.gms.maps.CameraUpdateFactory;
+import com.google.android.gms.maps.GoogleMap;
+import com.google.android.gms.maps.OnMapReadyCallback;
+import com.google.android.gms.maps.SupportMapFragment;
+import com.google.android.gms.maps.model.LatLng;
+import com.google.android.gms.maps.model.MarkerOptions;
+import com.google.android.gms.maps.model.PolylineOptions;
+
+import java.util.ArrayList;
+import java.util.EmptyStackException;
+import java.util.concurrent.ExecutionException;
+
+public class MapsActivity extends FragmentActivity implements OnMapReadyCallback, ILocationChange, ActivityCompat.OnRequestPermissionsResultCallback {
+
+    private GoogleMap mMap;
+    private Location currentLocation;
+    private static final int MY_PERMISSION = 123;
```

Comment:
Use proper ordering of variables, all the static final should be at one place, non static at one place rather than mixing them up in between